test/extensions/filters/http/cache/cache_filter_test.cc

```diff
@@ -575,6 +582,39 @@ TEST_F(CacheFilterTest, GetRequestWithBodyAndTrailers) {
   }
 }
 
+// Check that per route config is used in filter (in case if it is defined)
+TEST_F(CacheFilterTest, PerFilterConfigOverride) {
+  // Define per route config
+  const auto yaml_config = R"EOF(
+allowed_vary_headers:
+  - exact: bar
+typed_config:
+  "@type": "type.googleapis.com/envoy.extensions.cache.simple_http_cache.v3.SimpleHttpCacheConfig"
+)EOF";
+
+  envoy::extensions::filters::http::cache::v3::CacheConfig cache_per_route;
+  TestUtility::loadFromYaml(yaml_config, cache_per_route);
+
+  auto filter = makeFilter(nullptr);
+  CacheFilterConfig route_cfg(simple_cache_, cache_per_route);
+
+  ON_CALL(*decoder_callbacks_.route_, mostSpecificPerFilterConfig("envoy.filters.http.cache"))
+      .WillByDefault(Return(&route_cfg));
+
+  request_headers_.setHost("SomeHost");
+
+  // Filter cache should be null (due to we create filter with null cache - 'makeFilter(nullptr)')
+  ASSERT_TRUE(getFilterCache(filter) == nullptr);
+  EXPECT_EQ(Http::FilterHeadersStatus::StopAllIterationAndWatermark,
+            filter->decodeHeaders(request_headers_, true));
+
+  // During header decoding filter should reinit config using per route config.
+  // It means that cache should be initialized with 'simple_cache_' and allowed_vary_headers should
+  // be taken from per route config as well
+  ASSERT_TRUE(getFilterCache(filter) == simple_cache_);
```

Comment:
`ASSERT_EQ(getFilterCache(filter), simple_cache_);` would potentially have more useful output in case of failure. Also, probably makes more sense as EXPECT_EQ since the test isn't going to crash if the line fails, which would allow for *also* getting more useful output from the next line.